proxy/proxy.go

```diff
@@ -1099,12 +1054,6 @@ func (p *Proxy) do(ctx *context) error {
 
 	processedFilters := p.applyFiltersToRequest(ctx.route.Filters, ctx)
 
-	// per route rate limit
-	if settings, retryAfter := p.checkRatelimit(ctx); retryAfter > 0 {
```

Comment:
With this change, global rate limits are handled as proxy errors while route ratelimits are handled as regular response with 429 code. Not sure if it is important.